ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/FilterParameterTest.java

```diff
@@ -224,6 +243,91 @@ public void emptyListAsSubTopicDecodesCorrectly() throws JsonProcessingException
             emptyList());
   }
 
+  @Test
+  public void emptyListOfTopicsDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter = readJsonAsFilterParameter("{\"topics\":[]}");
+
+    assertThat(filterParameter.getTopics().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void emptyListOfTopicsNestedDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter = readJsonAsFilterParameter("{\"topics\":[[]]}");
+
+    assertThat(filterParameter.getTopics()).containsExactly(List.of());
+  }
+
+  @Test
+  public void singleTopicDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter("{\"topics\":[\"" + TOPIC_TWO + "\"]}");
+
+    assertThat(filterParameter.getTopics())
+        .containsExactly(List.of(LogTopic.fromHexString(TOPIC_TWO)));
+  }
+
+  @Test
+  public void singleNestedTopicDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter("{\"topics\":[[\"" + TOPIC_TWO + "\"]]}");
+
+    assertThat(filterParameter.getTopics())
+        .containsExactly(List.of(LogTopic.fromHexString(TOPIC_TWO)));
+  }
+
+  @Test
+  public void twoTopicsDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter("{\"topics\":[\"" + TOPIC_TWO + "\", \"" + TOPIC_THREE + "\"]}");
+
+    assertThat(filterParameter.getTopics())
+        .containsExactly(
+            List.of(LogTopic.fromHexString(TOPIC_TWO)),
+            List.of(LogTopic.fromHexString(TOPIC_THREE)));
+  }
+
+  @Test
+  public void twoTopicsInFirstPositionDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter(
+            "{\"topics\":[[\"" + TOPIC_TWO + "\", \"" + TOPIC_THREE + "\"]]}");
+
+    assertThat(filterParameter.getTopics())
+        .containsExactly(
+            List.of(LogTopic.fromHexString(TOPIC_TWO), LogTopic.fromHexString(TOPIC_THREE)));
+  }
+
+  @Test
+  public void nullInFirstPositionDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter("{\"topics\":[null, \"" + TOPIC_THREE + "\"]}");
+
+    assertThat(filterParameter.getTopics())
+        .containsExactly(singletonList(null), List.of(LogTopic.fromHexString(TOPIC_THREE)));
+  }
+
+  @Test
+  public void twoTopicsInFirstAndSecondPositionDecodesCorrectly() throws java.io.IOException {
+    final FilterParameter filterParameter =
+        readJsonAsFilterParameter(
+            "{\"topics\":["
```

Comment:
nit: is it overkill to use a StringJoiner to make this pretty?

Reply:
so my original problem was that I couldn't read the tests.... this makes it pretty explicit, I can simplify it down and reduce that if you like...

Reply:
new StringJoiner("[", "\",\", "]") might do kinda what you want? (then do a joiner.build)